AUTOMATAFORMAT.md

```diff
@@ -0,0 +1,111 @@
+  * `&`,`|`,`!`,`(`,`)` occurring as tokens in transition lines are logical operators.
+  * `[`,`]`,`-` within intervals of symbols of the form `[bla-bli]`.
+  * `{`,`}` are enclosing attributes, if we use them this way. Remains to be decided. (?)
+  * `@` opens the line with a section name and is used in transducer alphabet tokens of the form `x@a1`, `y@[10-55]`,...
+  * `%` opens a key-value line, possibly also key-value appendix of a transition. Remains to be decided. (?)
+  * `"` strings containing white spaces and reserved characters can be written in between `"`. The reserved characters lose their special meaning. The characters `"` and `\` inside a string must be escaped, i.e. `\"` and `\\`.
+  * `/` to concatenate lines.
+  * `\` for escaping characters.
+  * `a`,`q`,`n`,`f` are token type specifiers (alphabet, state, node, formula). If you don't like these, propose other ones.
+* The parser should recognise the reserved symbols as standalone tookens only if they appear in the special context where their special meaning applies, otehrwise they should be treated as normal symbols (is this reasonable ?).
+
+We categorise automata by **transition type**, that is, the structure of their transitions, and by **alphabet type**, i.e., how alphabet symbols on transitions are represented. It is expected that different parameters would be parsed into different data structures. These parameters should determine the name of the automata section.
+
+## Automata by transition type: 
+We use general form of AFA, by Pavol V., and simple NFA.
+
+### AFA 
+* Each state has a transition, a booelan formula over symbol literals, states, and nodes.
+* Several lines starting with the same state mean the disjunciton of the formulae. No line for the state means false. 
+* Since in the general formulae, the type of a token cannot be determined by its position, we require one of the three schemes for distinguishing token types.
+  * By **enumeration**.
+  * By **type-markers**. The literal is preceeded by a single character that specifies the type, we use `q` for states, `a` for symbols, and `n` for shared nodes. The marker is *not* a part of the thing's name! (Or should it be? It would perhaps be simpler that way?)
+  * **automatic**. If all the other types are specified by enumeration or by markers, the third type may be marked as automatic, the parser will assign it to everything which is not of the two previous types. 
+* The typing scheme is specified as a key value line with the key  `<Thing>-<Scheme>` where  `<Thing>` may be `Alphabet`, `States`, `Nodes`, or `Formulas` (Formulas will be explained later), and `<Scheme>` may be `markerd`, `enum`, or `auto`.  The cheme z `enum` is followed by an enumeration.  For instance, we can have `%Alphabet-enum a b c karel`, `%Alphabet-auto`, `%Alphabet-markers`.
+* Markers is the implicit option for all types of tokens, used if nothing is specified.
+* `()` means true and `!()` means false.
```

Comment:
what? this seems very exotic

Reply:
But does not need any new special character. The logic is () is a formula that says nothing (for all variables from the empty set, nothing holds ... ), which is true ....
Alternatives?

Reply:
`true` and `false` are reserved words